react_app/src/components/App.js

```diff
@@ -46,23 +66,22 @@ class App extends Component {
   }
 
   render() {
-    let content;
-
-    if (this.state.isAuthenticated) {
-      content = <GameContainer />;
-    } else {
-      if (this.state.newUser) {
-        content = <SignUpGizmo />;
-      } else {
-        content = <LoginGizmo />;
-      }
-    }
-
     return (
       <div className="App">
         <AppContext.Provider value={this.state}>
           <TitleBar title={"pew"} />
-          {content}
+
+          {
+            // eventually you'll want this little chunk here to
+            // go off and live in another component of its own IMO
+          }
+          {this.state.isAuthenticated ? (
+            <GameContainer />
+          ) : this.state.newUser ? (
+            <SignUpGizmo />
+          ) : (
+            <LoginGizmo />
+          )}
```

Comment:
I'd come to the same conclusion, if you look at the file tree changes I've started edging towards separating auth off into its own tree